core/controllers/galleries.py

```diff
@@ -225,22 +225,40 @@ def get(self):
 
 
 class ExplorationSummariesHandler(base.BaseHandler):
-    """Returns summaries corresponding to ids of public explorations."""
+    """Returns summaries corresponding to ids of public explorations. This
+    controller supports returning private explorations for the given user.
+    """
 
     def get(self):
         """Handles GET requests."""
         try:
             exp_ids = json.loads(self.request.get('stringified_exp_ids'))
         except Exception:
             raise self.PageNotFoundException
+        include_private_exps_str = self.request.get(
+            'include_private_explorations')
+        include_private_exps = (
+            include_private_exps_str.lower() == 'true'
+            if include_private_exps_str else False)
+
+        editor_user_id = self.user_id if include_private_exps else None
+        if not editor_user_id:
```

Comment:
(In fact, you may not even need the branching -- the line 253-256 call works in all cases.)